### ISSUE TITLE
Out of Memory

### ISSUE THREAD

**Issue by pribadihcr** [amplab/SparkNet#47]
Hi,

I have spark cluster with:
Workers: 5
Cores: 5 Total
Memory: 10.9 GB Total

When I run CifarApp, I got the following error:
ERROR ActorSystemImpl: Uncaught fatal error from thread [sparkDriver-akka.actor.default-dispatcher-3] shutting down ActorSystem [sparkDriver]
java.lang.OutOfMemoryError: Java heap space

many thank for help

**Comment by robertnishihara** [comment]
Hi,

You can increase the Java heap space by running `export _JAVA_OPTIONS="-Xmx3g"` on the master (and you can use a larger number than 3). Can you try that?

How much memory do you have on the Spark master? It's possible that there actually is not enough to store CIFAR in memory.

**Comment by pribadihcr** [comment]
@robertnishihara 

I have tried it.
I have 12G on spark master. 
So, How much minimum memory required on master, and each workers ?

**Comment by pribadihcr** [comment]
@robertnishihara 

I have tried it (export _JAVA_OPTIONS="-Xmx3g"). 
I have 12G memory master.
I tried run again with 5 workers. Memory each worker is 2.9G. And still same error.

How much minimum memory required on master and worker to run AppCifar example?

**Comment by robertnishihara** [comment]
Let me see if I can recreate the problem. In the meantime, do you know precisely where in the code it is running out of memory? In particular, is it during preprocessing? Perhaps the call to `val loader = new CifarLoader(sparkNetHome + "/caffe/data/cifar10/")`?

You might try commenting out everything after that line to figure out where the error is happening.

**Comment by robertnishihara** [comment]
I started a 5-worker Spark cluster on EC2. The workers are t2.small instances, which have 2GB memory, and the master is a t2.large instance, which has 8GB memory.

On this cluster, CifarApp works just fine for me. Note that the t2 instances don't have GPUs, so to get it working, I had to comment out the two calls to `caffeLib.set_mode_gpu()` in `Net.scala`.

Let me know where the app is crashing, then we can figure out what to try next.
